src/shared/tasks/sentry/tasksentry.cpp

```diff
@@ -76,12 +76,20 @@ void TaskSentry::run() {
 
 void TaskSentry::sendRequest() {
   NetworkRequest* request = new NetworkRequest(this, 200);
```

Comment:
Won't this allocation be leaked if the code returns at line 86? Should this line be moved below the new code block?

Reply:
Not in this case - `NetworkRequest` is self-destructive after the request has finished :) 
https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/shared/networkrequest.h#L26

Reply:
Thanks for pointing out the following in NetworkRequest will free the allocation:
connect(&m_timer, &QTimer::timeout, this, &NetworkRequest::maybeDeleteLater);
connect(m_reply, &QNetworkReply::finished, this,
          &NetworkRequest::maybeDeleteLater);